src/Maestro/Maestro.Web.Tests/SubscriptionsController20200220Tests.cs

```diff
@@ -162,6 +172,33 @@ public async Task GetAndListNonexistentSubscriptions()
             }
         }
 
+        [Test]
+        public async Task CreateSubscriptionForNonMicrosoftUserFails()
+        {
+            string testChannelName = "test-channel-sub-controller20200220";
+            string defaultGitHubSourceRepo = "https://github.com/dotnet/sub-controller-test-source-repo";
+            string defaultGitHubTargetRepo = "https://github.com/dotnet/sub-controller-test-target-repo";
+            string defaultBranchName = "main";
+            string anInvalidDependencyFlowNotificationList = "@someexternaluser;@somemicrosoftuser;@some-team";
+
+            // Create two subscriptions
+            SubscriptionData subscription = new SubscriptionData()
+            {
+                ChannelName = testChannelName,
+                Enabled = true,
+                SourceRepository = defaultGitHubSourceRepo,
+                TargetRepository = defaultGitHubTargetRepo,
+                Policy = new Api.v2018_07_16.Models.SubscriptionPolicy() { Batchable = true, UpdateFrequency = Api.v2018_07_16.Models.UpdateFrequency.EveryWeek },
+                TargetBranch = defaultBranchName,
+                PullRequestFailureNotificationTags = anInvalidDependencyFlowNotificationList
+            };
+
+            IActionResult result = await data.SubscriptionsController.Create(subscription);
+            result.Should().BeAssignableTo<BadRequestObjectResult>();
```

Comment:
I believe you can chain this like
```suggestion
            result.Should().BeAssignableTo<BadRequestObjectResult>().And.StatusCode.Should().Be((int) HttpStatusCode.BadRequest);
```

although I don't know if we really need to assert that "BadRequest" result returns "BadRequest". Its probably enough to assert that it returned a BadRequestObjectResult and call it a day.

Reply:
Removed three instances.